### PR TITLE
Use target context for removal of a single declaration

### LINKED ISSUE #3288: "Remove unused declaration" removes one NEWLINE token too many
Hi. My first report ever :-) 

## Example 1
Code Inspections - Variable oUserForm is not used - (above) `Fix` - Remove Unused Declaration would from code:
```
Public Function ProcessTextBox(ctrl As control) As String
    
'    Dim SearchRange As Range
    Dim FoundCell As Range
    Dim ws As Worksheet
    Dim word As String
    Dim rowNum As Long
    Dim ret As String
    Dim ctrlID As String
    Dim oUserForm As UserForm
    
    If resourceSheet Is Nothing Then Initialize
```
Remove `oUserForm` line but even the line under it
```
Public Function ProcessTextBox(ctrl As control) As String
    
'    Dim SearchRange As Range
    Dim FoundCell As Range
    Dim ws As Worksheet
    Dim word As String
    Dim rowNum As Long
    Dim ret As String
    Dim ctrlID As String
    If resourceSheet Is Nothing Then Initialize
```

## Example 2
Code Inspections - Variable myRib is not used - (above) `Fix` - Remove Unused Declaration would from code:
```
Option Explicit

Private myRib As IRibbonUI

'---------------------------------------------------------------------------------------
' Method : onAction callback
' Author : jverner
' Date   : 10.08.2017
' Purpose: ACTIONS Buttons
'---------------------------------------------------------------------------------------
Sub onAction(control As IRibbonControl, Optional ByVal pressed As Boolean)

    Select Case control.ID
```
Remove empty line bellow and all comments:
```
Option Explicit

Sub onAction(control As IRibbonControl, Optional ByVal pressed As Boolean)

    Select Case control.ID
```

### PR DESCRIPTION
**Pull request by Vogel612**
Additionally includes some unit tests for minor edge cases
Fixes #3288

### CODE REVIEW

File: Rubberduck.Parsing/Rewriter/RewriterInfo/VariableRewriterInfoFinder.cs [hunk 1]
```diff
@@ -61,9 +61,72 @@ private static RewriterInfo GetRewriterInfo(VBAParser.VariableSubStmtContext var
...
+                //if (nextStatement is ParserRuleContext ctx)
+                //{
+                //    tokenIndex = ctx.Start.TokenIndex - 1;
+                //}
+                //else
+                //{
+                //    // it makes no sense that the next statment would not be a ParserRuleContext
+                //    return RewriterInfo.None;
+                //}
+
```
Comment (retailcoder) [inline Rubberduck.Parsing/Rewriter/RewriterInfo/VariableRewriterInfoFinder.cs]: Dead code?
